File: eRent/eRent.Services/Database/Property.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace eRent.Services.Database
{
    public class Property
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [MaxLength(2000)]
        public string? Description { get; set; }

        [Required]
        public decimal PricePerMonth { get; set; } // Monthly rental price

        public decimal? PricePerDay { get; set; } // Daily rental price (optional, only if AllowDailyRental is true)

        public bool AllowDailyRental { get; set; } = false; // Whether landlord allows daily rentals

        [Required]
        public int Bedrooms { get; set; }

        [Required]
        public int Bathrooms { get; set; }

        [Required]
        public decimal Area { get; set; } // in square meters

        [Required]
        public int PropertyTypeId { get; set; }

        [Required]
        public int CityId { get; set; }

        [Required]
        public int LandlordId { get; set; } // FK to User (landlord)

        [MaxLength(200)]
        public string? Address { get; set; }

        [Required]
        public decimal Latitude { get; set; }

        [Required]
        public decimal Longitude { get; set; }

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public PropertyType PropertyType { get; set; } = null!;
        public City City { get; set; } = null!;
        public User Landlord { get; set; } = null!;

        // Navigation property for many-to-many relationship with Amenity
        public ICollection<PropertyAmenity> PropertyAmenities { get; set; } = new List<PropertyAmenity>();

        // Navigation property for PropertyImages
        public ICollection<PropertyImage> PropertyImages { get; set; } = new List<PropertyImage>();

        // Navigation property for Rents
        public ICollection<Rent> Rents { get; set; } = new List<Rent>();
    }
}

File: Startupba/UI/erent_desktop/lib/model/property.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:erent_desktop/model/amenity.dart';
import 'package:erent_desktop/model/property_image.dart';

part 'property.g.dart';

@JsonSerializable()
class Property {
  final int id;
  final String title;
  final String? description;
  final double pricePerMonth;
  final double? pricePerDay;
  final bool allowDailyRental;
  final int bedrooms;
  final int bathrooms;
  final double area;
  final int propertyTypeId;
  final String propertyTypeName;
  final int cityId;
  final String cityName;
  final int landlordId;
  final String landlordName;
  final String? address;
  final double latitude;
  final double longitude;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final List<Amenity> amenities;
  final List<PropertyImage> images;

  const Property({
    this.id = 0,
    this.title = '',
    this.description,
    this.pricePerMonth = 0.0,
    this.pricePerDay,
    this.allowDailyRental = false,
    this.bedrooms = 0,
    this.bathrooms = 0,
    this.area = 0.0,
    this.propertyTypeId = 0,
    this.propertyTypeName = '',
    this.cityId = 0,
    this.cityName = '',
    this.landlordId = 0,
    this.landlordName = '',
    this.address,
    this.latitude = 0.0,
    this.longitude = 0.0,
    this.isActive = true,
    required this.createdAt,
    this.updatedAt,
    this.amenities = const [],
    this.images = const [],
  });

  factory Property.fromJson(Map<String, dynamic> json) => _$PropertyFromJson(json);
  Map<String, dynamic> toJson() => _$PropertyToJson(this);
}
